client/blocks/term-form-dialog/index.jsx

```diff
@@ -244,13 +240,13 @@ class TermFormDialog extends Component {
 	}
 
 	renderParentSelector() {
-		const { labels, siteId, taxonomy, translate, terms } = this.props;
+		const { labels, siteId, taxonomy, translate, terms, term } = this.props;
 		const { isTopLevel, searchTerm, selectedParent } = this.state;
 		const query = {};
 		if ( searchTerm && searchTerm.length ) {
 			query.search = searchTerm;
 		}
-		const hideTermAndChildren = get( this.props.term, 'ID' );
+		const hideTermAndChildren = term?.ID;
```

Comment:
Should this be more obviously a boolean? We seem to use it as a `boolean` but it will either be `undefined` or a `number`, which seems off to me.

Reply:
Added `!!` in https://github.com/Automattic/wp-calypso/pull/57822/commits/5eed0c17a5940b415e3a7509ec585cacf8955eef